README.md
<!--
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages).

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages).
-->

## Features

一款简单易好用的刻度尺控件，自定义刻度显示，自定义样式

![](tutieshi_580x180_13s.gif)

![](tutieshi_580x188_8s.gif)

![](tutieshi_580x200_14s.gif)

## Getting started

## Usage

```dart
RulerWidget(
size: Size(size.width, size.height / 5),
viewStyle: ViewStyle.valueUpScaleDownCloseStyle,
showHighScaleLine: true,
showHighScaleNum: true,
showLowScaleLine: true,
showLowScaleNum: true,
showMiddleScaleLine: true,
showMiddleScaleNum: true,
showLine: true,
textSize: 12,
zoomTextSize: 23,
startValue: 3,
endValue: 30,
initValue: 15,
space: 20,
scaleValueSpace: 10,
lowScaleLineColor: Colors.white,
middleScaleLineColor: Colors.amber,
highScaleLineColor: Colors.red,
lineColor: Colors.white,
lowScaleLineHigh: 30,
middleScaleLineHigh: 40,
highScaleLineHigh: 44,
middleScaleLineStrokeWidth: 2,
lowScaleLineStrokeWidth: 2,
highScaleLineStrokeWidth: 2,
lineStrokeWidth: 2,
textColor: Colors.white,
middleSpaceValueSpace: 5,
highSpaceValeSpace: 10,
)
```

| 属性                         | 类型        | 描述                                   |
| -------------------------- | --------- | ------------------------------------ |
| size                       | Size      | view尺寸                               |
| viewStyle                  | ViewStyle | 显示样式                                 |
| showHighScaleLine          | boolen    | 是否显示高刻度线                             |
| showHighScaleNum           | boolen    | 是否显示高刻度值                             |
| showLowScaleLine           | boolen    | 是否显示低刻度线                             |
| showLowScaleNum            | boolen    | 是否显示低刻度值                             |
| showMiddleScaleLine        | boolen    | 是否显示中刻度线                             |
| showMiddleScaleNum         | boolen    | 是否显示中刻度值                             |
| showLine                   | boolen    | 是否显示中间横线                             |
| textSize                   | num       | 刻度值文字大小                              |
| zoomTextSize               | num       | 刻度值放大文字大小                            |
| startValue                 | int       | 起始值                                  |
| endValue                   | int       | 结束值                                  |
| initValue                  | int       | 初始值                                  |
| space                      | double    | 刻度间距                                 |
| scaleValueSpace            | double    | 刻度和刻度值之间的间隔距离                        |
| lowScaleLineColor          | Color     | 低刻度线条颜色                              |
| middleScaleLineColor       | Color     | 中刻度线条颜色                              |
| highScaleLineColor         | Color     | 高刻度线条颜色                              |
| lineColor                  | Color     | 中间横线颜色                               |
| lowScaleLineHigh           | double    | 低刻度线高度                               |
| middleScaleLineHigh        | double    | 中刻度线高度                               |
| highScaleLineHigh          | double    | 高刻度线高度                               |
| middleScaleLineStrokeWidth | double    | 中刻度线宽度                               |
| lowScaleLineStrokeWidth    | double    | 低刻度线宽度                               |
| highScaleLineStrokeWidth   | double    | 高刻度线宽度                               |
| lineStrokeWidth            | double    | 中间线条宽度                               |
| textColor                  | Color     | 刻度值文字颜色                              |
| middleSpaceValueSpace      | int       | 相邻中刻度值间隔                             |
| highSpaceValeSpace         | int       | 相邻高刻度值间隔 |


CHANGELOG.md
## 0.0.2

* 添加scale值回调


test/scale_ruler_view_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:scale_ruler_view/scale_ruler_view.dart';

void main() {
  test('adds one to input values', () {
  });
}


lib/scale_ruler_view.dart
import 'dart:ui';
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:flutter/widgets.dart';

class ViewStyle {
  int viewStyle = 0;

  ViewStyle(this.viewStyle);

  static ViewStyle valueDownScaleUpStyle = ViewStyle(1);

  static ViewStyle valueUpScaleDownStyle = ViewStyle(2);

  static ViewStyle valueDownScaleUpCloseStyle = ViewStyle(3);

  static ViewStyle valueUpScaleDownCloseStyle = ViewStyle(4);
}

typedef ScaleValueChangeCallback = void Function(int value);

class RulerWidget extends StatefulWidget {
  RulerWidget({
    Key? key,
    required this.size,
    required this.viewStyle,
    required this.startValue,
    required this.endValue,
    required this.initValue,
    required this.scaleValueSpace,
    required this.lowScaleLineHigh,
    required this.middleScaleLineHigh,
    required this.highScaleLineHigh,
    required this.space,
    required this.textSize,
    required this.textColor,
    required this.zoomTextSize,
    required this.lowScaleLineColor,
    required this.middleScaleLineColor,
    required this.highScaleLineColor,
    required this.lowScaleLineStrokeWidth,
    required this.middleScaleLineStrokeWidth,
    required this.highScaleLineStrokeWidth,
    required this.showLowScaleLine,
    required this.showMiddleScaleLine,
    required this.showHighScaleLine,
    required this.showLowScaleNum,
    required this.showMiddleScaleNum,
    required this.showHighScaleNum,
    required this.highSpaceValeSpace,
    required this.middleSpaceValueSpace,
    required this.lineColor,
    required this.showLine,
    required this.lineStrokeWidth,
    required this.callback,
  }) : super(key: key);
  Size size;
  var startValue = 3;
  var endValue = 30;
  var initValue = 10;
  double scaleValueSpace = 20;
  double lowScaleLineHigh = 20;
  double middleScaleLineHigh = 40;
  double highScaleLineHigh = 50;
  late ViewStyle viewStyle;
  double space = 60;
  double textSize = 14;
  Color textColor = Colors.black45;
  double zoomTextSize = 20;
  Color lowScaleLineColor;
  Color middleScaleLineColor;
  Color highScaleLineColor;
  double lowScaleLineStrokeWidth;
  double middleScaleLineStrokeWidth;
  double highScaleLineStrokeWidth;
  bool showLowScaleLine;
  bool showMiddleScaleLine;
  bool showHighScaleLine;
  bool showLowScaleNum;
  bool showMiddleScaleNum;
  bool showHighScaleNum;
  int middleSpaceValueSpace;
  int highSpaceValeSpace;
  bool showLine;
  double lineStrokeWidth;
  Color lineColor;
  ScaleValueChangeCallback callback;

  @override
  _RulerWidgetState createState() => _RulerWidgetState();
}

class _RulerWidgetState extends State<RulerWidget> with TickerProviderStateMixin {
  late double _leftXAxis;
  late double _rightXAxis;

  late AnimationController _controller;

  @override
  Widget build(BuildContext context) {
    _leftXAxis = -widget.space * (widget.initValue - widget.startValue);
    _rightXAxis = widget.space * (widget.endValue - widget.initValue);

    return GestureDetector(
      child: CustomPaint(
        size: widget.size,
        painter: RulerView(
          offset: _offsetX,
          textSize: widget.textSize,
          zoomTextSize: widget.zoomTextSize,
          middleScaleLineHigh: widget.middleScaleLineHigh,
          lowScaleLineHigh: widget.lowScaleLineHigh,
          highScaleLineHigh: widget.highScaleLineHigh,
          initValue: widget.initValue,
          startValue: widget.startValue,
          endValue: widget.endValue,
          scaleValueSpace: widget.scaleValueSpace,
          space: widget.space,
          viewStyle: widget.viewStyle,
          textColor: widget.textColor,
          lowScaleLineColor: widget.lowScaleLineColor,
          middleScaleLineColor: widget.middleScaleLineColor,
          highScaleLineColor: widget.highScaleLineColor,
          lowScaleLineStrokeWidth: widget.lowScaleLineStrokeWidth,
          middleScaleLineStrokeWidth: widget.middleScaleLineStrokeWidth,
          highScaleLineStrokeWidth: widget.highScaleLineStrokeWidth,
          showHighScaleLine: widget.showHighScaleLine,
          showMiddleScaleLine: widget.showMiddleScaleLine,
          showLowScaleLine: widget.showLowScaleLine,
          showMiddleScaleNum: widget.showLowScaleLine,
          showHighScaleNum: widget.showHighScaleNum,
          showLowScaleNum: widget.showLowScaleNum,
          middleSpaceValueSpace: widget.middleSpaceValueSpace,
          highSpaceValeSpace: widget.highSpaceValeSpace,
          lineColor: widget.lineColor,
          lineStrokeWidth: widget.lineStrokeWidth,
          callback: widget.callback,
          showLine: widget.showLine,
        ),
      ),
      onPanDown: (DragDownDetails e) {
        _onTouchLocationX = e.globalPosition.dx;
      },
      onPanUpdate: (DragUpdateDetails e) {
        setState(() {
          _offsetX = e.globalPosition.dx - _onTouchLocationX + _lastMoveX;
        });
      },
      onPanEnd: (DragEndDetails e) {
        _lastMoveX = _offsetX;

        startAnimation();
      },
    );
  }

  double _offsetX = 0;
  double _lastMoveX = 0;
  double _onTouchLocationX = 0;
  late Animation<double> _animation;

  void startAnimation() {
    double offset;
    if (_offsetX > 0 && _offsetX > _leftXAxis.abs()) {
      //超出左边界的偏移
      offset = _offsetX - _leftXAxis.abs();
    } else if (_offsetX < 0 && _offsetX < -_rightXAxis) {
      //超出右边界的偏移
      offset = (_rightXAxis.abs() - _offsetX.abs());
    } else {
      //刻度尺内的偏移
      offset = calculateOffset();
    }
    _controller = AnimationController(
      duration: const Duration(milliseconds: 200),
      vsync: this,
    );
    _animation = Tween(begin: 0.0, end: offset).animate(_controller)
      ..addListener(() {
        setState(() => {_offsetX = _lastMoveX - _controller.value * offset});
        debugPrint("startAnimation controller ${_controller.value}");
      })
      ..addStatusListener((status) {
        if (status == AnimationStatus.completed) {
          _lastMoveX = _offsetX;
        }
      });
    //启动动画(正向执行)
    _controller.forward();
  }

  double calculateOffset() {
    debugPrint("calculateOffset _offsetX ${_offsetX}");
    debugPrint("calculateOffset  widget.space : ${widget.space}");

    if (widget.space > _offsetX.abs()) {
      debugPrint("calculateOffset  calculateOffset : ${_offsetX}");
      return _offsetX;
    }
    debugPrint("calculateOffset  calculateOffset : ${_offsetX % widget.space}");
    return _offsetX % widget.space;
  }
}

class RulerView extends CustomPainter {
  late Paint linePaint;
  late Paint lowScalePaint;
  late Paint middleScalePaint;
  late Paint highScalePaint;
  int startValue;
  int endValue;
  int initValue;
  double scaleValueSpace;
  double space;
  var lowScaleLineHigh;
  var middleScaleLineHigh;
  var highScaleLineHigh;
  ViewStyle viewStyle;
  double zoomTextSize = 0;
  double textSize = 0;
  double offset = 0;
  Color textColor;
  Color lowScaleLineColor;
  Color middleScaleLineColor;
  Color highScaleLineColor;
  double lowScaleLineStrokeWidth;
  double middleScaleLineStrokeWidth;
  double highScaleLineStrokeWidth;
  bool showLowScaleLine;
  bool showMiddleScaleLine;
  bool showHighScaleLine;
  bool showLowScaleNum;
  bool showMiddleScaleNum;
  bool showHighScaleNum;
  int middleSpaceValueSpace;
  int highSpaceValeSpace;
  bool showLine;
  double lineStrokeWidth;
  Color lineColor;
  ScaleValueChangeCallback callback;

  RulerView(
      {required this.offset,
      required this.textSize,
      required this.zoomTextSize,
      required this.middleScaleLineHigh,
      required this.highScaleLineHigh,
      required this.scaleValueSpace,
      required this.endValue,
      required this.startValue,
      required this.initValue,
      required this.space,
      required this.textColor,
      required this.viewStyle,
      required this.lowScaleLineHigh,
      required this.lowScaleLineColor,
      required this.middleScaleLineColor,
      required this.highScaleLineColor,
      required this.lowScaleLineStrokeWidth,
      required this.middleScaleLineStrokeWidth,
      required this.highScaleLineStrokeWidth,
      required this.showLowScaleLine,
      required this.showMiddleScaleLine,
      required this.showHighScaleLine,
      required this.showLowScaleNum,
      required this.showMiddleScaleNum,
      required this.middleSpaceValueSpace,
      required this.highSpaceValeSpace,
      required this.showHighScaleNum,
      required this.lineColor,
      required this.lineStrokeWidth,
      required this.callback,
      required this.showLine}) {
    linePaint = Paint()
      ..style = PaintingStyle.fill
      ..color = this.lineColor
      ..strokeWidth = this.lineStrokeWidth;
    lowScalePaint = Paint()
      ..style = PaintingStyle.fill
      ..color = this.lowScaleLineColor
      ..strokeWidth = this.lowScaleLineStrokeWidth;
    middleScalePaint = Paint()
      ..style = PaintingStyle.fill
      ..color = this.middleScaleLineColor
      ..strokeWidth = this.middleScaleLineStrokeWidth;
    highScalePaint = Paint()
      ..style = PaintingStyle.fill
      ..color = this.highScaleLineColor
      ..strokeWidth = this.highScaleLineStrokeWidth;
  }

  @override
  void paint(Canvas canvas, Size size) {
    drawScale(canvas, size);
  }

  @override
  bool shouldRepaint(covariant CustomPainter oldDelegate) {
    return true;
  }

  void drawCoordinateAxes(Canvas canvas, Size size) {
    canvas.save();
    canvas.drawLine(Offset(0, -size.height / 2), Offset(0, size.height / 2), linePaint);
    canvas.drawLine(Offset(-size.width / 2, 0), Offset(size.width / 2, 0), linePaint);
    canvas.restore();
  }

  void drawScale(Canvas canvas, Size size) {
    canvas.save();
    if (viewStyle.viewStyle == ViewStyle.valueDownScaleUpCloseStyle.viewStyle || viewStyle.viewStyle == ViewStyle.valueUpScaleDownCloseStyle.viewStyle) {
      canvas.translate(size.width / 2, size.height / 2);
    } else if (viewStyle.viewStyle == ViewStyle.valueDownScaleUpStyle.viewStyle || viewStyle.viewStyle == ViewStyle.valueUpScaleDownStyle.viewStyle) {
      canvas.translate(size.width / 2, 0);
    }
    //绘制坐标轴
    // drawCoordinateAxes(canvas, size);
    num currentXAxisCoordinate = 0 + offset;
    int currentValue = initValue;
    while (currentXAxisCoordinate >= -size.width / 2 && currentValue >= startValue) {
      drawScaleItem(canvas, currentXAxisCoordinate, currentValue, size);
      _drawTextPaintShowSize(canvas, currentXAxisCoordinate, currentValue, size);
      currentXAxisCoordinate = currentXAxisCoordinate - space;
      currentValue = currentValue - 1;
    }
    currentXAxisCoordinate = 0 + offset;
    currentValue = initValue;
    while (currentXAxisCoordinate <= size.width / 2 && currentValue <= endValue) {
      drawScaleItem(canvas, currentXAxisCoordinate, currentValue, size);
      _drawTextPaintShowSize(canvas, currentXAxisCoordinate, currentValue, size);
      currentXAxisCoordinate = currentXAxisCoordinate + space;
      currentValue = currentValue + 1;
    }
    canvas.restore();
  }

  void drawScaleItem(Canvas canvas, num currentXAxisCoordinate, int currentValue, Size size) {
    var offsetLowStart;
    var offsetLowEnd;
    var offsetMiddleStart;
    var offsetMiddleEnd;
    var offsetHighStart;
    var offsetHighEnd;
    var offsetLineStart;
    var offsetLineEnd;
    if (viewStyle.viewStyle == ViewStyle.valueDownScaleUpCloseStyle.viewStyle) {
      offsetLowStart = Offset(currentXAxisCoordinate.toDouble(), -scaleValueSpace / 2);
      offsetLowEnd = Offset(currentXAxisCoordinate.toDouble(), -lowScaleLineHigh - scaleValueSpace / 2);
      offsetMiddleStart = Offset(currentXAxisCoordinate.toDouble(), -scaleValueSpace / 2);
      offsetMiddleEnd = Offset(currentXAxisCoordinate.toDouble(), -middleScaleLineHigh - scaleValueSpace / 2);
      offsetHighStart = Offset(currentXAxisCoordinate.toDouble(), -scaleValueSpace / 2);
      offsetHighEnd = Offset(currentXAxisCoordinate.toDouble(), -highScaleLineHigh - scaleValueSpace / 2);
      offsetLineStart = Offset(-size.width / 2, -scaleValueSpace / 2);
      offsetLineEnd = Offset(size.width / 2, -scaleValueSpace / 2);
    } else if (viewStyle.viewStyle == ViewStyle.valueUpScaleDownCloseStyle.viewStyle) {
      offsetLowStart = Offset(currentXAxisCoordinate.toDouble(), scaleValueSpace / 2);
      offsetLowEnd = Offset(currentXAxisCoordinate.toDouble(), lowScaleLineHigh + scaleValueSpace / 2);
      offsetMiddleStart = Offset(currentXAxisCoordinate.toDouble(), scaleValueSpace / 2);
      offsetMiddleEnd = Offset(currentXAxisCoordinate.toDouble(), middleScaleLineHigh + scaleValueSpace / 2);
      offsetHighStart = Offset(currentXAxisCoordinate.toDouble(), scaleValueSpace / 2);
      offsetHighEnd = Offset(currentXAxisCoordinate.toDouble(), highScaleLineHigh + scaleValueSpace / 2);
      offsetLineStart = Offset(-size.width / 2, scaleValueSpace / 2);
      offsetLineEnd = Offset(size.width / 2, scaleValueSpace / 2);
    } else if (viewStyle.viewStyle == ViewStyle.valueDownScaleUpStyle.viewStyle) {
      offsetLowStart = Offset(currentXAxisCoordinate.toDouble(), lowScaleLineHigh.toDouble());
      offsetLowEnd = Offset(currentXAxisCoordinate.toDouble(), 0);
      offsetMiddleStart = Offset(currentXAxisCoordinate.toDouble(), middleScaleLineHigh.toDouble());
      offsetMiddleEnd = Offset(currentXAxisCoordinate.toDouble(), 0);
      offsetHighStart = Offset(currentXAxisCoordinate.toDouble(), highScaleLineHigh.toDouble());
      offsetHighEnd = Offset(currentXAxisCoordinate.toDouble(), 0);
      offsetLineStart = Offset(-size.width / 2, 0);
      offsetLineEnd = Offset(size.width / 2, 0);
    } else if (viewStyle.viewStyle == ViewStyle.valueUpScaleDownStyle.viewStyle) {
      offsetLowStart = Offset(currentXAxisCoordinate.toDouble(), size.height);
      offsetLowEnd = Offset(currentXAxisCoordinate.toDouble(), size.height - lowScaleLineHigh);
      offsetMiddleStart = Offset(currentXAxisCoordinate.toDouble(), size.height);
      offsetMiddleEnd = Offset(currentXAxisCoordinate.toDouble(), size.height - middleScaleLineHigh);
      offsetHighStart = Offset(currentXAxisCoordinate.toDouble(), size.height);
      offsetHighEnd = Offset(currentXAxisCoordinate.toDouble(), size.height - highScaleLineHigh);
      offsetLineStart = Offset(-size.width / 2, size.height);
      offsetLineEnd = Offset(size.width / 2, size.height);
    }
    if (showLine) {
      canvas.drawLine(offsetLineStart, offsetLineEnd, linePaint);
    }
    if (showLowScaleLine) {
      canvas.drawLine(offsetLowStart, offsetLowEnd, lowScalePaint);
    }
    if (currentValue % middleSpaceValueSpace == 0) {
      if (showMiddleScaleLine) {
        canvas.drawLine(offsetMiddleStart, offsetMiddleEnd, middleScalePaint);
      }
    }
    if (currentValue % highSpaceValeSpace == 0) {
      if (showHighScaleLine) {
        canvas.drawLine(offsetHighStart, offsetHighEnd, highScalePaint);
      }
    }
  }

  int callbackCache = -1;
  void _drawTextPaintShowSize(Canvas canvas, num currentXAxisCoordinate, int currentValue, Size canvasSize) {
    // if (currentXAxisCoordinate == 0 || currentXAxisCoordinate.abs() < space * 0.3) {
    if (currentXAxisCoordinate == 0){
    if (callback != null) {
        callbackCache = currentValue;
        callback.call(currentValue);
      }
    }
    double finalTextSize = textSize;
    if (currentXAxisCoordinate.abs() <= space) {
      if (currentXAxisCoordinate == 0) {
        finalTextSize = zoomTextSize;
      } else {
        finalTextSize = (((space - currentXAxisCoordinate.abs()) / space) * (zoomTextSize - textSize)) + textSize;
      }
    } else {
      finalTextSize = textSize;
    }
    TextPainter textPainter =
        TextPainter(text: TextSpan(text: '$currentValue', style: TextStyle(fontSize: finalTextSize, color: textColor)), textAlign: TextAlign.center, textDirection: TextDirection.ltr);
    textPainter.layout(); // 进行布局
    Size size = textPainter.size; // 尺寸必须在布局后获取
    var offset;
    if (viewStyle.viewStyle == ViewStyle.valueDownScaleUpCloseStyle.viewStyle) {
      offset = Offset(currentXAxisCoordinate - size.width / 2, 0 + scaleValueSpace / 2);
    } else if (viewStyle.viewStyle == ViewStyle.valueUpScaleDownCloseStyle.viewStyle) {
      offset = Offset(currentXAxisCoordinate - size.width / 2, 0 - scaleValueSpace / 2 - size.height);
    } else if (viewStyle.viewStyle == ViewStyle.valueDownScaleUpStyle.viewStyle) {
      offset = Offset(currentXAxisCoordinate - size.width / 2, canvasSize.height - size.height);
    } else if (viewStyle.viewStyle == ViewStyle.valueUpScaleDownStyle.viewStyle) {
      offset = Offset(currentXAxisCoordinate - size.width / 2, 0);
    }
    if (((currentValue % middleSpaceValueSpace == 0 && showMiddleScaleNum) || (currentValue % highSpaceValeSpace == 0 && showHighScaleNum) || showLowScaleNum)) textPainter.paint(canvas, offset);
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:scale_ruler_view/scale_ruler_view.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);
  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    final size = MediaQuery.of(context).size;
    return Scaffold(
        appBar: AppBar(
          title: Text(widget.title),
        ),
        body: Container(
          child: Center(
              child: Container(
                color: Colors.black87,
                child:
                RulerWidget(
                  size: Size(size.width, size.height / 5),
                  viewStyle: ViewStyle.valueUpScaleDownCloseStyle,
                  showHighScaleLine: true,
                  showHighScaleNum: true,
                  showLowScaleLine: true,
                  showLowScaleNum: true,
                  showMiddleScaleLine: true,
                  showMiddleScaleNum: true,
                  showLine: true,
                  textSize: 12,
                  zoomTextSize: 23,
                  startValue: 3,
                  endValue: 30,
                  initValue: 15,
                  space: 20,
                  scaleValueSpace: 10,
                  lowScaleLineColor: Colors.white,
                  middleScaleLineColor: Colors.amber,
                  highScaleLineColor: Colors.red,
                  lineColor: Colors.white,
                  lowScaleLineHigh: 30,
                  middleScaleLineHigh: 40,
                  highScaleLineHigh: 44,
                  middleScaleLineStrokeWidth: 2,
                  lowScaleLineStrokeWidth: 2,
                  highScaleLineStrokeWidth: 2,
                  lineStrokeWidth: 2,
                  textColor: Colors.white,
                  middleSpaceValueSpace: 5,
                  highSpaceValeSpace: 10,
                  callback:(int value){
                    debugPrint("scaleValueChange  ${value}");
                  },
                ),
              )
          ),
        )
      // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}
